app/questionnaire/path_finder.py

```diff
@@ -33,22 +33,22 @@ def __init__(
         self.list_store = list_store
         self.supplementary_data_store = supplementary_data_store
 
-    def routing_path(
-        self, section_id: str, list_item_id: str | None = None
-    ) -> RoutingPath:
+    def routing_path(self, section_key: SectionKey) -> RoutingPath:
         """
         Visits all the blocks in a section and returns a path given a list of answers.
         """
         routing_path_block_ids: list[str] = []
-        current_location = Location(section_id=section_id, list_item_id=list_item_id)
-        section = self.schema.get_section(section_id)
+        current_location = Location(
+            section_id=section_key.section_id, list_item_id=section_key.list_item_id
```

Comment:
yeah `Location(*section_key)` would result in `Location(section_id=section_key.section_id, block_id=section_key.list_item_id)`
but you could unpack it with correct kwargs by doing `Location(**section_key._asdict())`

Reply:
Not a fan of using the private method in this case.

Reply:
Came across it in [this answer](https://stackoverflow.com/questions/26180528/convert-a-namedtuple-into-a-dictionary)  mentioning its a [documented method](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._asdict), and underscore is just for avoiding conflicts rather than indicating it being private.

That said, Its not all that readable, so I agree with not doing it - just for interest

Reply:
@katie-gardner @Yuyuutsu There are a few places where unpacking would be neater, what if we added our own public `asdict()` wrapper?.

Reply:
Sure - I'd be all for that

Reply:
Yeah that sounds good

Reply:
Done in [`b1f5f77` (#1196)](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1196/commits/b1f5f77c9d66c50882f668f7a4c9607eccc4dd06)